src/vstart.sh

```diff
@@ -57,13 +57,29 @@ export DYLD_LIBRARY_PATH=$CEPH_LIB:$DYLD_LIBRARY_PATH
 [ -z "$CEPH_NUM_FS"  ] && CEPH_NUM_FS="$FS"
 [ -z "$CEPH_NUM_RGW" ] && CEPH_NUM_RGW="$RGW"
 
-[ -z "$CEPH_NUM_MON" ] && CEPH_NUM_MON=3
-[ -z "$CEPH_NUM_OSD" ] && CEPH_NUM_OSD=3
-[ -z "$CEPH_NUM_MDS" ] && CEPH_NUM_MDS=3
-[ -z "$CEPH_NUM_MGR" ] && CEPH_NUM_MGR=1
 [ -z "$CEPH_NUM_FS"  ] && CEPH_NUM_FS=1
-[ -z "$CEPH_MAX_MDS" ] && CEPH_MAX_MDS=1
 [ -z "$CEPH_NUM_RGW" ] && CEPH_NUM_RGW=0
+[ -z "$CEPH_MAX_MDS" ] && CEPH_MAX_MDS=1
+
+# setup a new cluster with the default number of daemons if none of the
+# CEPH_NUM_* is specified.
+if [ -z "$CEPH_NUM_MON" -a \
+     -z "$CEPH_NUM_OSD" -a \
+     -z "$CEPH_NUM_MDS" -a \
+     -z "$CEPH_NUM_MGR" ]; then
+    CEPH_NUM_MON=3
+    CEPH_NUM_OSD=3
+    CEPH_NUM_MDS=3
+    CEPH_NUM_MGR=1
+    kill_all=1
+else
+    # and don't start any other daemons if any CEPH_NUM_* is specified.
+    [ -z "$CEPH_NUM_MON" ] && CEPH_NUM_MON=0
+    [ -z "$CEPH_NUM_OSD" ] && CEPH_NUM_OSD=0
+    [ -z "$CEPH_NUM_MDS" ] && CEPH_NUM_MDS=0
+    [ -z "$CEPH_NUM_MGR" ] && CEPH_NUM_MGR=0
+    kill_all=0
+fi
```

Comment:
i usually just start MON and OSD, without MDS or MGR, this saves the time to ready and launch these daemons, and also i don't compile MDS by default.